services/ingestion/etl/inforegio/xml/src/lib/transform.js

```diff
@@ -81,25 +81,40 @@ const getAddress = record => {
 };
 
 /**
- * Formats information for the `value` of {BudgetItem}
+ * Formats information for the {BudgetItem}
  *
  * @memberof inforegioXmlTransform
  * @param {string} budget Prefixed currency value
- * @returns {number} The value for `value` of {BudgetItem}
+ * @returns {BudgetItem} The formatted budget
  *
  * @example
  * input => "EUR 329 000 000"
  * output => "329000000"
  */
 const formatBudget = budget => {
-  if (!budget || typeof budget !== 'string') return 0;
+  if (!budget || typeof budget !== 'string')
+    return { value: 0, currency: '', raw: '' };
+
+  let formattedCurrency = budget.split(' ')[0];
+
+  // Convert 'ECU' to 'EUR'
+  if (formattedCurrency === 'ECU') {
+    formattedCurrency = 'EUR';
+  }
 
   const formattedBudget = budget
     .split(' ')
     .slice(1)
     .join('');
 
-  return Number(formattedBudget) || 0;
+  return {
+    value: Number(formattedBudget) || 0,
+    currency:
+      formattedCurrency && formattedCurrency.length === 3
```

Comment:
The `length` rule will skip some actual currencies, such as the `FF`
```
"eu_contrib": {
  "value": 3200000,
  "currency": "",
  "raw": "FF 3 200 000"
}
```

In my opinion, it's better to have the currency value at this stage, even if the enrichment ignores it. It'll surely be better than having the `value` only and `raw` with something, but no currency.

Reply:
One of Laurent's requirements was: 

> Use ISO codes for currencies

What we can do is to keep a list of mappings between two-letter codes and their normalized three-letter version.

Examples:
FF -> FRF
DM -> DEM


But I would never store a non-normalized currency here